calculator/main.py

```diff
@@ -26,7 +31,14 @@ def perform_operation(calc, operation, params):
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
     """
-    pass
+    if all([isinstance(num, int) or isinstance(num, float) for num in params]):
```

Comment:
Instead of your or statement here `isinstance(num, (int, float))` accomplishes the same thing while being more concise.